vignettes/OrganismDbi.Rmd

```diff
@@ -0,0 +1,190 @@
+is the `r Biocpkg("Homo.sapiens")` package, which combines access to the two
+resources above along with others.
+
+This is made possible because the packages that are represented by 
+`r Biocpkg("Homo.sapiens")`
+are related to each other via foreign keys.
+
+```{r figure, fig.cap = "Relationships between Annotation packages", echo = FALSE}
+knitr::include_graphics("relationships-between-annotation-packages.png")
+```
+
+#   Getting started with OrganismDbi
+
+Usage of a package like this has been deliberately kept very simple. The methods
+supported are the same ones that work for all the packages based on
+*AnnotationDb* objects. The methods that can be applied to these new packages
+are `columns`, `keys`, `keytypes` and `select`. 
+
+So to learn which kinds of data can be retrieved from a package like this we
+would simply load the package and then call the `columns` method.
+
+```{r columns, message = FALSE, warning = FALSE}
+library(Homo.sapiens)
+columns(Homo.sapiens)
+```
+
+To learn which of those kinds of data can be used as keys to extract data, we
+use the `keytypes` method.
+
+```{r keys1, message = FALSE, warning = FALSE}
+keytypes(Homo.sapiens)
+```
+
+To extract specific keys, we need to use the `keys` method, and also provide it
+a legitimate keytype:
+
+```{r keys2, message = FALSE, warning = FALSE}
+head(keys(Homo.sapiens, keytype = "ENTREZID"))
+```
+
+And to extract data, we can use the `select` method. The select method depends
+on the values from the previous three methods to specify what it will extract.
+Here is an example that will extract, UCSC transcript names, and gene symbols
+using Entrez Gene IDs as keys.
+
+```{r select, message = FALSE, warning = FALSE}
+k <- head(keys(Homo.sapiens, keytype = "ENTREZID"), n = 3)
+select(
+  Homo.sapiens,
+  keys = k,
+  columns = c("TXNAME", "SYMBOL"),
+  keytype = "ENTREZID"
+)
+```
+
+In addition to `select`, some of the more popular range based methods have also
+been updated to work with an *AnnotationDb* object. So for example you could
+extract transcript information like this:
+
+```{r transcripts, message = FALSE, warning = FALSE}
+transcripts(Homo.sapiens, columns = c("TXNAME", "SYMBOL"))
+```
+
+And the *GRanges* object that would be returned would have the information that
+you specified in the columns argument. You could also have used the `exons` or
+`cds` methods in this way.
+
+The `transcriptsBy`,`exonsBy` and `cdsBy` methods are also supported. For
+example:
+
+```{r transcriptsBy, message = FALSE, warning = FALSE}
+transcriptsBy(Homo.sapiens,
+              by = "gene",
+              columns = c("TXNAME", "SYMBOL"))
+```
+
+#   Making your own OrganismDbi packages
+
+So in the preceding section you can see that using an 
+`r Biocpkg("OrganismDbi")`
+package behaves very similarly to how you might use a `r Biocpkg("TxDb")` or an
+`r Biocpkg("OrgDb")` package. The same methods are defined, and they behave
```

Comment:
On lines 111,112 and 129, TxDb and OrgDb aren't packages. I believe they're kinds of packages, so let's just mark them with backticks.